src/core/qgsrelationmanager.cpp

```diff
@@ -21,16 +21,13 @@
 #include "qgsvectordataprovider.h"
 #include "qgsvectorlayer.h"
 
-QgsRelationManager::QgsRelationManager( QgsProject *project )
-  : QObject( project )
+QgsRelationManager::QgsRelationManager( QgsProject &project )
```

Comment:
In general QObject based members are passed by pointer (not reference), just like Qt does too.

N.b. it would be nice if the server would follow this approach too.

Reply:
Hi @m-kuhn 
No problem to sacrifice to Qt standards but I'm always uncomfortable with the fact that you may pass null pointer when non null pointer is required and appreciate static typing for this - this why we use reference instead.

By the way, this is qgis core changes not only server.

Reply:
Do a `Q_ASSERT( pointer )` and document it in the API to make your uncomfortability explicit.

Reply:
@m-kuhn  note that this is a pointer thas is passed to QObject - not the reference.

Reply:
Reverted signature in order to prevent API break. Still allow nullptr to preserve previous behavior.